apps/core/tests/test_event_loop_runner.py

```
@@ -0,0 +1,113 @@
+    def _latest_error_record(self, level: int) -> logging.LogRecord | None:
+        for record in reversed(self.capture_handler.records):
+            if record.levelno >= level:
+                return record
+        return None
+
+    def test_handle_command_exception_logs_context(self):
+        self._register_command_handler()
+        self.capture_handler.records.clear()
+        command = _DummyCommand({})
+
+        with self.assertRaises(RuntimeError):
```

Ruff (0.14.6)

Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)